### PR TITLE
Log viewer tail refresh

### PR DESCRIPTION
**Pull request by gportela85**
https://github.com/exhi/hoist-react/issues/330

### CODE REVIEW

File: admin/tabs/logging/viewer/LogViewerDisplay.js [hunk 1]
```diff
@@ -47,6 +55,17 @@ class LogViewerDisplay extends Component {
         });
     }
 
+    timedRefreshLines = () => {
```
Comment (lbwexler) [inline admin/tabs/logging/viewer/LogViewerDisplay.js]: Total nit -- use method here.  Reserving these fat arrow properties for event handlers in render methods.

File: admin/tabs/logging/viewer/LogViewerDisplay.js [hunk 2]
```diff
@@ -47,6 +55,17 @@ class LogViewerDisplay extends Component {
...
+        const {model, lastRow} = this,
+            {tabPaneModel, tail} = model,
+            rect = lastRow.value.getBoundingClientRect(),
+            outOfView = rect.bottom > window.innerHeight;
```
Comment (lbwexler) [inline admin/tabs/logging/viewer/LogViewerDisplay.js]: can't lastRow be null if the file is empty (or just hasn't been rendered).  We should make this null save.

**Comment by lbwexler** [comment]
Had one remaining change -- regarding null safety 

As discussed, this whole pattern of having to pass around and consult tabPaneModel to see if we are 'visible' seems wonky, but we have it in a number of places.  Could we establish a concept of visibility that HoistComponent could easily read from context!?  TabPanes could then write to that, so that all children could see whether they were globally visible or not....  This might also help us to cleanup our story around TabContainer itself.

**Comment by amcclain** [comment]
Sweet thank you guys.

**Pull request merged**